src/p_maputl.c

```diff
@@ -444,6 +444,9 @@ boolean P_BlockThingsIterator(int x, int y, boolean func(mobj_t*))
     if (func == PIT_RadiusAttack || func == PIT_ChangeSector)
       return true;
 
+    if (mobj->flags2 & MF2_RIP)
```

Comment:
Don't we have to check the `tmthing`'s flags?

Reply:
Right, this is wrong approach.